test/src/main/java/org/axonframework/test/aggregate/AggregateTestFixture.java

```diff
@@ -166,6 +167,12 @@ public AggregateTestFixture(Class<T> aggregateType) {
         return this;
     }
 
+    @Override
+    public FixtureConfiguration<T> useStateStorage() {
+        this.useStateStorage = true;
```

Comment:
I think that with the addition of the `boolean`, it is good to check whether it's `true` for any `given()` invocations (so the method which assumes event sourcing).
Throwing an `AxonConfigurationException` when it is true on this functions would be best.

Reply:
I added a check to the `andGiven()` method and changed `givenNoPriorActivity()` so that it will no longer call `andGiven()`.

Reply:
As per discussion in the main thread, we'll revamp the `givenNoPriorActivity` to be useable for both event-sourcing and state-storage. To that end, the current provided adjustments suffice, with the exception of the current Javadoc format of the `givenNoPriorActivity` method.